### PR TITLE
WIP Fixes accessibility behavior for topic management in the repository home view

### PR DESCRIPTION
**Pull request by fsologureng**
These changes address the impossibility of selecting the anchor for click "Manage topics" in the repository home view and, once inside the management widget, the impossibility of selecting each current topic for the action of delete.
Contributed by @forgejo.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: web_src/js/features/repo-home.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ export function initRepoTopicBar() {
     viewDiv.hide();
     editDiv.css('display', ''); // show Semantic UI Grid
   });
+  mgrBtn.on('keydown', (e) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      viewDiv.hide();
+      editDiv.css('display', ''); // show Semantic UI Grid
```
Comment (delvh) [inline web_src/js/features/repo-home.js:22]: We could think about extracting the function body (without condition) into a separate method…

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -2331,6 +2331,7 @@ tag.create_success = Tag '%s' has been created.
 
 topic.manage_topics = Manage Topics
 topic.done = Done
+topic.delete = Delete
```
Comment (delvh) [inline options/locale/locale_en-US.ini:2334]: `Delete topic`?
Or is it always clear for users what is supposed to be deleted?
```suggestion
topic.delete = Delete topic
```

File: web_src/js/features/repo-home.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-home.js:22]: Maybe use a `button`, then it can automatically handle `enter` as click.

**Comment by fsologureng** [comment]
I have realised that this solution is unusable under the current behaviour of the related dropdown menu; inside the topic manager the delete topic buttons are not clickable and the newly added items contain a different structure so the solution will not work.
Also I've noticed some strange bugs doing back and forward with TAB while adding and deleting topics.

I will leave it as WIP until I understand the problems and develop a complete solution.

File: web_src/js/features/repo-home.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-home.js:22]: Here is an example about how to use `button` correctly, instead of duplicating code again.

* #22807

**Comment by fsologureng** [comment]
Hi @wxiaoguang: can you suggest me a way to change the accessibility pattern of this dropdown, which is not a menu but a dropdown? The [global modification of the fomantic static template for the dropdown menu](https://github.com/go-gitea/gitea/blob/main/web_src/js/features/aria.js#L17) prevents me to make that change, because the role of the items is not `menuitem` but `option`. I have tried to change this modification by a callback function through the `Show` event, but the fomantic bind function of the event is not stackable (I mean, it doesn't support two calls), so I will have problems with [this kind of settings](https://github.com/go-gitea/gitea/blob/main/web_src/js/features/repo-issue-content.js#L55).

**Comment by wxiaoguang** [comment]
I can only answers some questions based on my previous experiences. I am not reading the code at the moment, just my opinion, for information only.

> a way to change the accessibility pattern of this dropdown?

If you can hijack the Fomantic UI's module, maybe you can change almost everything from it.

> which is not a menu but a combobox?

According to my previous tests, https://github.com/go-gitea/gitea/blob/main/web_src/js/features/aria.md , the `menuitem` seems working better, while `option` has various problems when it's used with the dropdown - it's hard to describe, I just tested and felt at that time. Maybe one of the problems is that Fomantic UI dropdown could be used as a menu container, and it could also be used as an option container. You can try to find out whether there are better approaches, for example, introduce more fixes to use `option` instead?

> The global modification of the fomantic static template for the dropdown menu prevents me to make that change ...  but the fomantic bind function of the event is not stackable

Maybe you could read the Fomantic UI code to see whether there are other approaches and make more fixes to the dropdown, or replace the Fomantic UI module with other components.

**Comment by wxiaoguang** [comment]
I think this issue might answer some of your questions. The root problem is that the hard-coded focus management inside Fomantic Dropdown. I think it's unfixable at the moment.

 [Proposal] Improve the dropdown step by step #23345

**Comment by wxiaoguang** [comment]
> because the role of the items is not menuitem but option

If you think it causes problem, you could exclude it from the `aria.js` (make `aria.js` skip some unwanted elements). According to my previous tests, it might not work well because of the "focus" problem. I am not sure.

----

I proposed a new PR (the best I can do at the moment).

* https://github.com/go-gitea/gitea/pull/23346

I think it could improve a11y a lot.

To resolve the screen reader problem fundamentally, IMO the component code should be rewritten ....

**Comment by wxiaoguang** [comment]
Closed by #23542 and #23626 (due to the comment https://github.com/go-gitea/gitea/pull/22631/files#r1089583380 was nerve resolved and it doesn't handle dynamically generated labels)

**Pull request closed without merging**